src/pages/Swap/AdvancedSwapMode/AdvancedSwapMode.component.tsx

```diff
@@ -0,0 +1,178 @@
+import { PropsWithChildren, useRef } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Flex, Text } from 'rebass'
+
+import { ButtonDark } from '../../../components/Button'
+import { useAdvancedTradingView } from '../../../services/AdvancedTradingView/useAdvancedTradingView.hook'
+import { useAllTrades } from '../../../services/AdvancedTradingView/useAllTrades.hook'
+import {
+  AdvancedModeHeader,
+  AdvancedModeTitle,
+  AdvancedSwapModeWrapper,
+  BaseWrapper,
+  ChartWrapper,
+  LiquidityWrapper,
+  OrderButton,
+  OrderHistoryHeader,
+  OrdersWrapper,
+  PairDetailsWrapper,
+  SwapBox,
+  SwitchButton,
+  SwitcherWrapper,
+  TradesWrapper,
+  TransactionsWrapper,
+} from './AdvancedSwapMode.styles'
+import { Chart } from './Chart'
+import { ColumnHeader } from './ColumnHeader'
+import { InfiniteScroll } from './InfiniteScroll'
+import { OrderHistoryTransaction } from './OrderHistory/OrderHistoryTransaction'
+import { PairDetails } from './PairDetails'
+
+export const AdvancedSwapMode = ({ children }: PropsWithChildren) => {
+  const { t } = useTranslation('swap')
+
+  const tradesWrapper = useRef<HTMLDivElement>(null)
+
+  const {
+    transactions,
+    tradeHistory,
+    liquidityHistory,
+    hasMore: { hasMoreTrades, hasMoreActivity },
+  } = useAllTrades()
+
+  const {
+    chainId,
+    inputToken,
+    outputToken,
+    symbol,
+    showTrades,
+    isLoadingTrades,
+    isLoadingActivity,
+    fetchTrades,
+    fetchActivity,
+    pairTokens: [token0, token1],
+    activeCurrencyOption,
+    handleAddLiquidity,
+    handleSwitchCurrency,
+    isFetched,
+  } = useAdvancedTradingView()
+
+  const transactionsWrapperMaxHeight = tradesWrapper?.current?.clientHeight
+    ? tradesWrapper?.current?.clientHeight - 85
+    : 570
```

Comment:
Should we give this variables a name?

Reply:
yea, I think we can.

Reply:
fixed [501266b](https://github.com/levelkdev/swapr-dapp/pull/1503/commits/501266b3e2ac4a76be7811d2fa67025ee4d1d58b)